src/store/modules/search.store.js

```diff
@@ -51,7 +55,29 @@ const actions = {
                 }
             }
 
-            if (coordinates) {
+            if (extractedCoordinate) {
+                let coordinates = [...extractedCoordinate.coordinate]
+                if (extractedCoordinate.coordinateSystem !== currentProjection) {
+                    // special case for LV03 input, we can't use proj4 to transform them into
+                    // LV95 or others, as the deformation between LV03 and the others is not constant.
+                    // So we pass through a LV95 reframe (done by a backend service that knows all deformations between the two)
+                    // and then go to the wanted coordinate system
+                    if (extractedCoordinate.coordinateSystem === LV03) {
+                        coordinates = await reframe({
+                            inputProjection: LV03,
+                            inputCoordinates: coordinates,
+                        })
+                        coordinates = proj4(LV95.epsg, currentProjection.epsg, coordinates).map(
+                            currentProjection.roundCoordinateValue
+                        )
```

Comment:
? I don't get it why using reframe and proj4 ?

Reply:
because if we are using something else than LV95, reframe only returns LV95 coordinates (say when we will be using mercator as the main projection)

Reply:
If I understood correctly the goal of using reframe was for user that used lv03 to lv95 and not lv03 to other projection ? Am I correct. Because converting from lv03 to lv95 to wgs84 or any other will anyway add some imprecision ?
Would it be not simpler and explicite to have something
```javascript
if (extractedCoordinate.coordinateSystem === LV03 && currentProjection.epsg == LV95) {
   coordinates = await reframe(...)
} else {
   coordinates = proj4(...)
}
```

Or at least in you current solution only call proj4 if the current projection is not LV95.

Reply:
the reframe backend service only deals with coordinates in LV03 or LV95 (output too) so proj4 is still required if we are aiming for anything else.
What requires special care (the dedicated backend) is the transformation between LV03 and LV95, as it's not a linear change

Reply:
calling proj4 on the same in and out will just returns the coordinates, that won't use any calculation (that's why I let it go through without checking)